### ISSUE TITLE
Be able to use one connection for multiple producer/consumer

### ISSUE THREAD

**Issue by MarcAntoine-Arnaud** [svix/omniqueue-rs#61]
Hi,

Thank you for this crate and job made !

I wanna to know if a reason exists to didn't have more attributes on [QueueBuilder](https://docs.rs/omniqueue/0.2.0/omniqueue/builder/struct.QueueBuilder.html) to pass queue parameters.
Based on that we can une one connection with multiple queues.

Or do you see another way to address that ?

Best,
Marc-Antoine

**Comment by svix-jplatte** [comment]
Yeah, this isn't currently possible. I'm thinking the best solution would be to allow users to pass in a connection pool instead of an address (but keep the address-based initialization as a convenience). This runs a bit counter to another goal of decoupling the public API from other crates like `redis`, but the best fix for that might be to split omniqueue into multiple independent crates anyways: one base crate for the traits, dynamic dispatch and such and one crate per backend like `omniqueue-redis` (which would then be the only crate to need a release when doing an upgrade of the `redis` dependency).
